misc/compiler-plugins/modules/configurable-schema-generator/src/main/java/org/ballerinalang/configschemagenerator/util/ConfigSchemaBuilder.java

```diff
@@ -0,0 +1,334 @@
+        // Create root node
+        JsonObject root = new JsonObject();
+        root.addProperty("$schema", "http://json-schema.org/draft-07/schema#");
+        root.addProperty(TYPE, "object");
+        root.add(PROPERTIES, rootNode);
+        for (Map.Entry<ConfigModuleDetails, List<ConfigVariable>> configModuleDetails : configDetails.entrySet()) {
+            List<ConfigVariable> configVariables = configModuleDetails.getValue();
+            if (isSingleFileProject(configModuleDetails)) {
+                // Set configurable variables at package level
+                setConfigVariables(configVariables, rootNode);
+                setRequiredConfigs(configVariables, root);
+            } else {
+                // Get org node
+                JsonObject orgNode;
+                JsonElement orgJsonNode = rootNode.get(configModuleDetails.getKey().orgName());
+                if (orgJsonNode == null) {
+                    orgNode = createObjNode(new JsonObject());
+                } else {
+                    orgNode = (JsonObject) orgJsonNode;
+                }
+                // Get package node
+                JsonObject pkgNode;
+                JsonObject orgPropertiesNode = (JsonObject) orgNode.get(PROPERTIES);
+                JsonElement pkgJsonNode = orgPropertiesNode.get(configModuleDetails.getKey().packageName());
+                if (pkgJsonNode == null) {
+                    pkgNode = createObjNode(new JsonObject());
+                    orgPropertiesNode.add(configModuleDetails.getKey().packageName(), pkgNode);
+                } else {
+                    pkgNode = (JsonObject) pkgJsonNode;
+                }
+                JsonObject packagePropertiesNode = (JsonObject) pkgNode.get(PROPERTIES);
+                // Set configurable variables
+                if (configModuleDetails.getKey().moduleName().equals(configModuleDetails.getKey().packageName())) {
+                    // Set configurable variables at package level
+                    setConfigVariables(configVariables, packagePropertiesNode);
+                    setRequiredConfigs(configVariables, pkgNode);
+                } else {
+                    JsonObject modVarNode = new JsonObject();
+                    JsonObject modObjNode = createObjNode(setConfigVariables(configVariables, modVarNode));
+                    setRequiredConfigs(configVariables, modObjNode);
+                    packagePropertiesNode.add(configModuleDetails.getKey().moduleName(), modObjNode);
+                }
+                rootNode.add(configModuleDetails.getKey().orgName(), orgNode);
+            }
+        }
+
+        return root;
+    }
+
+    /**
+     * Check if given module details are for a single file project.
+     *
+     * @param configModuleDetails key entry for config details map
+     * @return boolean flag
+     */
+    private static boolean isSingleFileProject(Map.Entry<ConfigModuleDetails,
+            List<ConfigVariable>> configModuleDetails) {
```

Comment:
We can modify the ConfigmoduleDetails to hold the ModuleDescriptor and check if it is a single file as follows:
`moduleDescriptor.packageDescriptor().name().value().equals(".") && moduleDescriptor.packageDescriptor().org().anonymous()`

Reply:
What if we add a flag to indicate whether it's a single file project to the `ConfigmoduleDetails`? That way we can only expose the required info. WDYT?

Reply:
Added the project kind to the `ConfigmoduleDetails` which can be used to check if single file project and also whether this module is in the current package with https://github.com/ballerina-platform/ballerina-lang/pull/33674